### ISSUE TITLE
Arrow keys stop working after exiting a shell opened via the terminal verb

### ISSUE THREAD

**Issue by therealkrispet** [Canop/broot#452]
After opening a new shell from broot via the terminal verb (or the ctrl+t shortcut) and closing it by sending EOF (pressing ctrl+d), the arrow keys no longer seem to work. I can still navigate around using the mouse or by searching for something (so broot is still responsive), but things that require the arrow keys to be pressed no longer seem to function. Curiously, when using the exit built-in to close the opened shell, this is not the case and the arrow keys work fine after.

- open broot via the br function
- select a directory by navigating to it via the arrow keys or searching
- press ctrl+t to open a shell in that directory (or call the :terminal verb)
- exit the shell by pressing ctrl+d (EOF)
- the arrow keys should no longer work in broot

Expected behaviour:
- the arrow keys should function fine after exiting a shell with EOF

**Configuration (please complete the following information):**
- broot version: broot 1.6.3
- OS: Arch Linux, updated just before reproducing the issue
- Version: latest
- no custom broot config

broot.log: [broot.log](https://github.com/Canop/broot/files/7254894/broot.log)

**Comment by Canop** [comment]
I've just tested with a few terminals on debian/gnome and I don't reproduce this problem. What terminal do you use ?

**Comment by therealkrispet** [comment]
Sorry, I've forgotten to include this in the original bug report, but I use Konsole as a terminal emulator and the shell is zsh. No multiplexers. I tested it with kitty as well and the same thing happens.
I'm starting to think that this has something to do with my zsh config or plugins, as I'm unable to reproduce the issue with bash (as in insert `export SHELL=/bin/bash` as the first step in the reproduction).
I keep my dotfies in git, so during installation I've just added the output of `broot --print-shell-function zsh` to my zshrc and then executed `broot --set-install-state installed`. Is this br install method correct or could this cause the issue?

**Comment by Canop** [comment]
I just tested with kitty and zsh (no plugin) with no problem.
I don't see how the way you installed `br` could be a problem but it may be easily checked by trying with `broot` instead of `br`.

**Comment by therealkrispet** [comment]
Yup, that's not it, the behaviour is the same when starting with `broot` instead of `br`.
I've tested this a bit more and I think that it might be caused by oh-my-zsh (it's a zsh configuration framework). 
It loads itself by doing a `source $ZSH/oh-my-zsh.sh` from your zshrc. The issue is no longer present after commenting out that line. If we could verify this somehow, then oh-my-sh could be appended/prepended to the name of this issue so that others may find it. It's a relatively popular framework.

**Comment by Canop** [comment]
I've just installed oh-my-zsh and tested, and didn't reproduce the problem. Maybe a specific plugin or config you use ?

**Comment by therealkrispet** [comment]
Now I'm testing with a single-line `.zshrc` file:
```
source "${HOME}/.oh-my-zsh/oh-my-zsh.sh"
```
With this, I can still reproduce the issue. After deleting this line and trying with a completely empty `.zshrc` the issue is no longer present. 

Entering the `${HOME}/.oh-my-zsh` directory and looking at the git repo there it seems that right now I have commit `2ea1fa6cbfc6340463906dfed589d5c976a10abb` checked out. It automatically updates by git pulling a few times a month after starting up a shell. I did this now manually and I have commit `54783f66b8188297448bc725c578fdc7647e9fa0` now, which should be the latest (https://github.com/ohmyzsh/ohmyzsh/commits/master). I can still reproduce the issue like this and it still disappears with an empty `.zshrc`.

**Comment by therealkrispet** [comment]
Also, could you tell me what debian distribution are you running (stable/testing/unstable)? I was thinking about spinning up a VM to test broot with my dotfiles there. I could find out if it's caused by some package behaving differently on a closer to upstream version or something like that.

**Comment by Canop** [comment]
I've tested your ctrl-t then ctrl-d problem on Debian 10 buster.

**Comment by therealkrispet** [comment]
This is starting to get weird. I've bootstrapped a minimal buster container to test this with.
I used systemd-nspawn, this is what I did:

```
sudo debootstrap --include=systemd-container stable buster/ http://deb.debian.org/debian/
sudo systemd-nspawn --resolv-conf=bind-host -D buster/ passwd        # change root password so I can log in
sudo systemd-nspawn --resolv-conf=bind-host -bD buster/                   # start container

apt update
apt install git curl wget zsh
sh -c "$(curl -fsSL https://raw.github.com/ohmyzsh/ohmyzsh/master/tools/install.sh)"    # install oh-my-zsh

# install broot
echo "deb [signed-by=/usr/share/keyrings/azlux-archive-keyring.gpg] http://packages.azlux.fr/debian/ buster main" | tee /etc/apt/sources.list.d/azlux.list
wget -O /usr/share/keyrings/azlux-archive-keyring.gpg  https://azlux.fr/repo.gpg
apt update && apt install broot
broot --install

br  # attempt reproduction
```

I was able to reproduce the issue in an environment like this, but only when using Konsole as a terminal emulator. If using kitty for instance, the issue no longer occours. Which is weird, because on my host arch system it seems to be independent of the terminal emulator, because there I can reproduce it using kitty as well.

The other weirdness is that no matter the environment, setting SHELL=/bin/bash or even /bin/sh makes the problem go away, so it may be related to how my terminal emulator interacts with zsh+oh-my-zsh (because disabling oh-my-zsh inhibits the issue as well). 

TL;DR I'm not quite sure what causes this and honestly I have no clue how I could fix it (yet), but I'm quite certain that it is caused by some weirdness in my specific config/setup and not a bug in broot. At least I cannot prove it, so I'm going to close this for now if that's OK with you. I'll comment again if I find a solution so it might help others, or reopen if it really turns out to be a bug. Thank you for your help and attempting to reproduce this!

**Comment by Canop** [comment]
Thanks for the investigation. And of course I'd welcome any new information on this topic.

**Comment by Canop** [comment]
And, btw, you might be interested by this part of my .bashaliases so that you don't use ctrl-d anymore:

```bash
alias :e=nvim
alias :q=exit
```

**Comment by therealkrispet** [comment]
Those aliases are a nice idea with all the vim muscle memory. I like these in broot too. Great project, by the way, glad to be using it. Thank you for the effort and the tip with the aliases as well!
